upstairs/src/lib.rs

```diff
@@ -2772,6 +2773,19 @@ impl Guest {
         Ok(())
     }
 
+    pub fn byte_offset_to_block(
+        &self,
+        offset: u64,
+    ) -> Result<Block, CrucibleError> {
+        let bs = self.query_block_size()?;
```

Comment:
Yes and no :)

Yes because there's no restriction in the type that function returns (it's just a u64)

Hopefully no because

```
pub struct RegionDefinition {
    /**
     * The size of each block in bytes. Must be a power of 2, minimum 512.
     */
    block_size: u64,
```
and
```
impl RegionOptions {
    pub fn validate(&self) -> Result<()> {
        if !self.block_size.is_power_of_two() {
            bail!("block size must be a power of two, not {}", self.block_size);
        }

        if self.block_size < 512 {
            bail!("minimum block size is 512 bytes, not {}", self.block_size);
        }
```